buildtest/cli/compilers.py

```diff
@@ -47,31 +47,44 @@ def compiler_test(configuration):
     Args:
         configuration (buildtest.config.SiteConfiguration): An instance of SiteConfiguration class
     """
+    pass_compilers = []
+    fail_compilers = []
+
     bc = BuildtestCompilers(configuration=configuration)
-    bc.find_compilers()
+    register_compilers = bc.names()
+
+    print("Find these registered compilers: ", register_compilers)
 
-    table = Table(title="Compilers Test Pass")
-    table.add_column("No.", style="cyan", no_wrap=True)
-    table.add_column("Compiler Name", style="green")
-    table.add_column("Status", justify="right")
+    for module in register_compilers:
+        cmd = Module(module, debug=True)
+        ret = cmd.test_modules(login=True)
+        if ret == 0:
+            pass_compilers.append(module)
+        else:
+            fail_compilers.append(module)
 
-    for compiler_cat in bc.compiler_modules_lookup:
-        for compiler in bc.compiler_modules_lookup[compiler_cat]:
-            table.add_row(str(table.row_count + 1), compiler, "✅")
+    if pass_compilers:
+
+        table = Table(title="Compilers Test Pass")
+        table.add_column("No.", style="cyan", no_wrap=True)
+        table.add_column("Compiler Name", style="green")
+        table.add_column("Status", justify="right")
+
+        for idx, pass_compiler in enumerate(pass_compilers):
+            table.add_row(str(idx + 1), pass_compiler, "✅")
 
-    if table.row_count:
```

Comment:
i think this if statement should stand if there is no rows to print then dont print the table. With this change you will print the table with 0 rows.